force_gromacs/data_sources/simulation/simulation_data_source.py

```diff
@@ -18,6 +20,15 @@ class SimulationDataSource(BaseDataSource):
 
     simulation_builder = Instance(GromacsSimulationBuilder)
 
+    def _check_perform_simulation(self, model, results_path):
+        """Check to see whether a simulation should be performed.
+        If a results file already exists, then only perform a new
+        simulation if required by the model"""
+
+        if os.path.exists(results_path):
+            return model.ow_data
+        return True
```

Comment:
What if the result file doesn't exist, but the model says "no simulation"?

Reply:
Good question, if `model.dry_run==True`, then this is currently dealt with later on by the `GromacsPipeline` class. In which case the directory tree for the simulation data is also created and a bash script produced, but the `GromacsPipeline` does not call any executable commands.